### PR TITLE
Digital zoom support

### PR DESCRIPTION
**Pull request by derFogel**
I added a simple zoom slider and zooming using the volume keys.

### CODE REVIEW

**Comment by piggz** [comment]
Nice, but i dont like bundling a binary with the source, it also wont work on non arm devices.  Could you do a PR with _just_ the slider, as I have an idea for utilising the volume keys.  Cheers.

**Comment by derFogel** [comment]
I removed the binaries. How do you want to utilize the volume keys. I tried using libresourceqt5 and installing an event filter (only works for the camera key on the Xperia X) but none of those worked.

If your way doesn't work I can modify the qmldir file to load the mediakey plugin from the default directory (I don't know why I haven't done this).

**Comment by derFogel** [comment]
Maybe libresourceqt5 and an event filter works. I haven't tried this.

**Comment by mlehtima** [comment]
Maybe check the example here how to use volume buttons https://github.com/nemomobile/libaudioresource/issues/1

**Comment by derFogel** [comment]
@mlehtima Thanks for your suggestion. I already tried using it in the past but it didn't work. Now it works but I don't know why.

**Comment by piggz** [comment]
@mlehtima that is the exact method i was going to use! :)

File: src/harbour-advanced-camera.cpp [hunk 1]
```diff
@@ -48,9 +49,15 @@ int main(int argc, char *argv[])
 
     QQuickView *view = SailfishApp::createView();
 
+    ResourceHandler *handler = new ResourceHandler();
```
Comment (piggz) [inline src/harbour-advanced-camera.cpp]: maybe just create on the stack?

File: src/harbour-advanced-camera.cpp [hunk 2]
```diff
@@ -48,9 +49,15 @@ int main(int argc, char *argv[])
...
+    handler->acquire();
```
Comment (piggz) [inline src/harbour-advanced-camera.cpp]: do we need to release at any point? destruction?

File: harbour-advanced-camera.pro [hunk 3]
```diff
@@ -68,4 +69,8 @@ HEADERS += \
     src/wbmodel.h \
     src/focusmodel.h \
     src/flashmodel.h \
-    src/fsoperations.h
+    src/fsoperations.h \
+    src/resourcehandler.h
+
+INCLUDEPATH += /usr/include/resource/qt5
```
Comment (piggz) [inline harbour-advanced-camera.pro]: Do we need to add anything to the .yaml/.spec?

**Comment by piggz** [comment]
Also, can you squash into a single commit?

File: harbour-advanced-camera.pro (hunk 3 above)
Comment (mlehtima) [reply harbour-advanced-camera.pro]: I think we should use pkg-config in both the make `libresourceqt5` a dependency. Same for this .pro file should use pkg-config to set needed things `CONFIG += link_pkgconfig
PKGCONFIG += libresourceqt5`. You can check https://github.com/nemomobile/qmlgallery how each part is done.

File: src/harbour-advanced-camera.cpp (hunk 2 above)
Comment (derFogel) [reply src/harbour-advanced-camera.cpp]: Yes we need to release when the up is minimized so you can adjust the volume.

File: harbour-advanced-camera.pro (hunk 3 above)
Comment (derFogel) [reply harbour-advanced-camera.pro]: Using libresourceqt5 is not possible since it is not allowed in the harbour.

**Comment by derFogel** [comment]
Since libresource(qt5) is not allowed in the harbour and libaudioresource doesn't allow grabbing keys I needed to do a trick. I link the app against libaudioresource and so against libresource (since libaudioresource requires libresource). Then I use dlsym to load the required functions. I know it's not the best solution but it works and doesn't requires dozens on libraries to be shipped.

**Comment by derFogel** [comment]
@piggz squashed. (Wasn't sure if there is a way to reverse it if i need to.)

**Pull request merged**